### PR TITLE
migrated to the "new" mongo driver

### PR DESCRIPTION
**Pull request by silvadanilo**

### CODE REVIEW

File: src/Recruiter/Infrastructure/Persistence/Mongodb/URI.php [hunk 1]
```diff
@@ -3,107 +3,46 @@
...
-            sprintf(
-                "Sorry, I don't recognize '%s' as valid MongoDB coordinates",
-                $uri
-            )
-        );
+        return new self($uri);
     }
 
-    private static function optionsFrom(string $queryString): array
+    public function database()
```
Comment (dbellettini) [inline src/Recruiter/Infrastructure/Persistence/Mongodb/URI.php]:
```suggestion
    public function database(): string
```

File: src/Recruiter/Job/Repository.php [hunk 2]
```diff
@@ -466,9 +466,10 @@ private function recentArchivedOrScheduledJobsWithManyAttempts(
     private function map($cursor)
     {
         $jobs = [];
-        while ($cursor->hasNext()) {
-            $jobs[] = Job::import($cursor->getNext(), $this);
+        foreach ($cursor->toArray() as $document) {
```
Comment (dbellettini) [inline src/Recruiter/Job/Repository.php]:
```suggestion
        foreach ($cursor as $document) {
```
should work

File: src/Recruiter/Worker/Repository.php [hunk 3]
```diff
@@ -45,24 +47,26 @@ public function deadWorkers($consideredDeadAt)
     {
         return $this->roster->find(
             ['last_seen_at' => [
-                '$lt' => new MongoDate($consideredDeadAt->getTimestamp())]
+                '$lt' => new MongoUTCDateTime($consideredDeadAt->format('U') * 1000)]
```
Comment (dbellettini) [inline src/Recruiter/Worker/Repository.php]: It's not well documented but MongoUTCDateTime can directly take a `DateTime` or `DateTimeImmutable` in its constructor

Comment (silvadanilo) [reply src/Recruiter/Worker/Repository.php]: but the result is different:
```
php > $datetime = new DateTime();
php > $x = new MongoDB\BSON\UTCDateTime($datetime);
php > $y = new MongoDB\BSON\UTCDateTime($datetime->format('U') * 1000);
php > 
php > var_export($d);
DateTime::__set_state(array(
   'date' => '2019-04-18 10:49:21.385441',
   'timezone_type' => 3,
   'timezone' => 'Europe/Berlin',
))
php > 
php > var_export($x);
MongoDB\BSON\UTCDateTime::__set_state(array(
   'milliseconds' => '1555577769571',
))
php > 
php > var_export($y);
MongoDB\BSON\UTCDateTime::__set_state(array(
   'milliseconds' => '1555577769000',
))
```

**Pull request closed without merging**